pyvrp/tests/test_ProblemData.py

```diff
@@ -351,3 +368,36 @@ def test_vehicle_type_attribute_access():
     assert_allclose(vehicle_type.fixed_cost, 3)
     assert_allclose(vehicle_type.tw_early, 17)
     assert_allclose(vehicle_type.tw_late, 19)
+
+
+def test_depot_client_location_access():
```

Comment:
It's great that you added this test but I'm not 100% sure about the interface. What is the benefit of having `.client()` that indexes from zero? I think it's currently used nowhere (or almost nowhere) in the code but it can cause a lot of confusion potentially.

I'd suggest either:
* not having `client(idx)` and then maybe also not have `depot(idx)` for consistency and just use `location(idx)` everywhere,
* or having `client(idx)` and `depot(idx)` using the 'absolute' index, i.e. just being convenience functions for `location(idx)`, but with a check on the `idx` (and, thus, a 'guarantee' on the returned type of the location).

I'm in favor of the second option:
* We could revert most `.location(client)` into `.client(client)` (if not all?) which is more consistent with using `.depot()` for indexing depots and additionally reduces the diff of this PR by quite a bit.
* Given that clients and depots are different things (we already have some 'conditions', i.e. prize/demand/service duration is 0), I can imagine a future where we'd strongly type them and I think this would be most compatible with that situation.

Reply:
I'm going for the first because it's easiest to implement, and keeps the indexing at least somewhat consistent. Indices that do not start from zero for separate things is a gotcha I'd like to avoid. Having everything in one function is somewhat consistent with e.g. `Model`.